huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account.go

```diff
@@ -0,0 +1,566 @@
+	bodyParams := map[string]interface{}{
+		"name":    utils.ValueIngoreEmpty(d.Get("name")),
+		"host":    utils.ValueIngoreEmpty(d.Get("host")),
+		"comment": utils.ValueIngoreEmpty(d.Get("description")),
+	}
+	param := map[string]interface{}{
+		"users": []interface{}{bodyParams},
+	}
+	return param
+}
+
+func buildUpdateGaussDBAccountPasswordBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":     utils.ValueIngoreEmpty(d.Get("name")),
+		"host":     utils.ValueIngoreEmpty(d.Get("host")),
+		"password": utils.ValueIngoreEmpty(d.Get("password")),
+	}
+	param := map[string]interface{}{
+		"users": []interface{}{bodyParams},
+	}
+	return param
+}
+
+func resourceGaussDBAccountDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// deleteGaussDBAccount: delete the GaussDB MySQL account
+	var (
+		deleteGaussDBAccountHttpUrl = "v3/{project_id}/instances/{instance_id}/db-users"
+		deleteGaussDBAccountProduct = "gaussdb"
+	)
+	deleteGaussDBAccountClient, err := cfg.NewServiceClient(deleteGaussDBAccountProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	deleteGaussDBAccountPath := deleteGaussDBAccountClient.Endpoint + deleteGaussDBAccountHttpUrl
+	deleteGaussDBAccountPath = strings.ReplaceAll(deleteGaussDBAccountPath, "{project_id}",
+		deleteGaussDBAccountClient.ProjectID)
+	deleteGaussDBAccountPath = strings.ReplaceAll(deleteGaussDBAccountPath, "{instance_id}", instanceID)
+
+	deleteGaussDBAccountOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	deleteGaussDBAccountOpt.JSONBody = utils.RemoveNil(buildDeleteGaussDBAccountBodyParams(d))
+
+	var deleteGaussDBAccountResp *http.Response
+	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutDelete), func() *resource.RetryError {
+		deleteGaussDBAccountResp, err = deleteGaussDBAccountClient.Request("DELETE",
+			deleteGaussDBAccountPath, &deleteGaussDBAccountOpt)
+		isRetry, err := handleGaussDBMysqlOperationError(err)
+		if isRetry {
+			return resource.RetryableError(err)
+		}
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return diag.Errorf("error deleting GaussDB MySQL account: %s", err)
+	}
+
+	deleteGaussDBAccountRespBody, err := utils.FlattenResponse(deleteGaussDBAccountResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jobId, err := jmespath.Search("job_id", deleteGaussDBAccountRespBody)
+	if err != nil {
+		return diag.Errorf("error deleting GaussDB MySQL account: job_id is not found in API response")
+	}
+
+	err = waitForJobComplete(ctx, deleteGaussDBAccountClient, d.Timeout(schema.TimeoutUpdate), instanceID, jobId.(string))
```

Comment:
```suggestion
	err = waitForJobComplete(ctx, deleteGaussDBAccountClient, d.Timeout(schema.TimeoutDelete), instanceID, jobId.(string))
```
this line is more than 120